### PR TITLE
IBP-3262 Set StudyDataManager Annotation

### PR DESCRIPTION
**Pull request by gelli-leafnode**
Add annotation to StudyDataManager `@Resource`. 
_StudyDataManager `@Resource` variables are not initialized / null during runtime. _

### CODE REVIEW

File: src/test/java/org/generationcp/middleware/service/impl/study/BMSAPIStudyServiceMock.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import org.generationcp.middleware.service.api.study.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.annotation.Resource;
+import java.util.List;
+@Service
+@Transactional
+public class BMSAPIStudyServiceMock implements StudyService {
```
Comment (darla-leafnode) [inline src/test/java/org/generationcp/middleware/service/impl/study/BMSAPIStudyServiceMock.java]: As discussed, we don't need to create this mock implementation for testing null value in autowiring

File: src/test/java/org/generationcp/middleware/service/impl/study/StudyServiceImplIntegrationTest.java [hunk 2]
```diff
@@ -96,4 +101,42 @@ public void testGetStudyDetailsForGeolocation() {
 		Assert.assertEquals(this.study.getName() + " Environment Number 1", studyDetailsDto.getMetadata().getStudyName());
 	}
 
+	@Test(expected = NullPointerException.class)
+	public void testGetStudyDetailsForGeolocationNull() {
```
Comment (darla-leafnode) [inline src/test/java/org/generationcp/middleware/service/impl/study/StudyServiceImplIntegrationTest.java]: We can remove this test as we delete the mock impl class too

File: src/main/java/org/generationcp/middleware/service/impl/study/StudyServiceImpl.java [hunk 3]
```diff
@@ -42,9 +42,11 @@
 import org.hibernate.type.StringType;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/service/impl/study/StudyServiceImpl.java]: Remove unused import

File: src/test/java/org/generationcp/middleware/service/impl/study/StudyServiceImplIntegrationTest.java [hunk 4]
```diff
@@ -96,4 +98,23 @@ public void testGetStudyDetailsForGeolocation() {
...
+			this.testDataInitializer
+				.createDmsProject("Summary Dataset", "Summary Dataset-Description", study, study, DatasetTypeEnum.SUMMARY_DATA);
+
+		final int locationId = 101;
+		final Geolocation geolocation = this.testDataInitializer.createTestGeolocation("1", locationId);
+		this.testDataInitializer
+			.createTestExperiment(environmentDataset, geolocation, TermId.TRIAL_ENVIRONMENT_EXPERIMENT.getId(), "0", null);
+		this.testDataInitializer.addProjectProp(study, TermId.PI_ID.getId(), "", VariableType.STUDY_DETAIL, "1", 6);
+		final StudyDetailsDto studyDetailsDto = this.studyService.getStudyDetailsForGeolocation(geolocation.getLocationId());
+		Assert.assertFalse(CollectionUtils.isEmpty(studyDetailsDto.getContacts()));
```
Comment (darla-leafnode) [inline src/test/java/org/generationcp/middleware/service/impl/study/StudyServiceImplIntegrationTest.java]: It would be stronger to add assertion on comparison of studyDetails.getContacts().get(0) to the user details retrieved from workbench users table.

Comment (clarysabel) [reply src/test/java/org/generationcp/middleware/service/impl/study/StudyServiceImplIntegrationTest.java]: @darla-leafnode @gelli-leafnode This test is failing in our local environments and also failed sometimes in jenkins. Please check and fix it

Comment (clarysabel) [reply src/test/java/org/generationcp/middleware/service/impl/study/StudyServiceImplIntegrationTest.java]: It is solved by: 
`		this.testDataInitializer.addProjectProp(this.study, TermId.PI_ID.getId(), "", VariableType.STUDY_DETAIL, String.valueOf(user.getPerson().getId()), 6);
`
Not sure how it was merged to master. Even today I run a branch build which I think merge with master first and it run fine. CI is running ok, so I am not so sure how it is working and if we should expend more time verifying if we have a problem running tests in jenkins

Comment (clarysabel) [reply src/test/java/org/generationcp/middleware/service/impl/study/StudyServiceImplIntegrationTest.java]: We will ignore it for now, if you have some time please check and fix it
